.github/workflows/uc-oss.yml

```diff
@@ -0,0 +1,77 @@
+name: uc-oss
+
+on:
+  pull_request: # for testing
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+    paths:
+      - .github/workflows/uc-oss.yml
+      - mlflow/protos/**
+      - mlflow/store/**
+  schedule:
+    # Run this workflow daily at 13:00 UTC
+    - cron: "0 13 * * *"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash --noprofile --norc -exo pipefail {0}
+
+env:
+  MLFLOW_HOME: /home/runner/work/mlflow/mlflow
+
+jobs:
+  uc-oss-integration-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+    permissions: {}
+    if: github.event_name == 'workflow_dispatch' || (github.event_name == 'schedule' && github.repository == 'mlflow-automation/mlflow') || (github.event_name == 'pull_request' && github.event.pull_request.draft == false)
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: ${{ github.event_name == 'schedule' && 'mlflow/mlflow' || github.event.inputs.repository }}
+          ref: ${{ github.event.inputs.ref }}
+          submodules: recursive
+      - uses: ./.github/actions/free-disk-space
+      - uses: ./.github/actions/setup-python
+      - uses: ./.github/actions/setup-pyenv
+
+      - name: Install dependencies
+        run: |
+          source ./dev/install-common-deps.sh --ml
+          pip install fastapi uvicorn tf-keras
+
+      - name: Set up Java 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: "temurin" # Use Temurin distribution of OpenJDK
+
+      - name: Clone UnityCatalog at tag v0.2.1
+        run: |
+          git clone --branch v0.2.1 --depth 1 https://github.com/unitycatalog/unitycatalog.git
+
+      - name: Build MLflow UI
+        working-directory: mlflow/server/js
+        run: |
+          yarn
+          yarn build
+
+      - name: Start uc-oss server
+        working-directory: unitycatalog
+        run: |
+          build/sbt package
+          nohup bin/start-uc-server > uc-server.log 2>&1 &
+
+      - name: Run tests for UnityCatalog
+        run: |
+          echo "Running tests for UnityCatalog..."
+          export PYTHONPATH=$MLFLOW_HOME:$PYTHONPATH
```

Comment:
Why do we need to modify `PYTHONPATH`?

Reply:
Copied this from another test.  Will remove and see if it still runs